chainstate/tx-verifier/src/transaction_verifier/mod.rs

```diff
@@ -315,25 +346,102 @@ where
+                            Some(res)
+                        }
+                        TxOutput::CreateStakePool(pool_id, _) => {
+                            // If the input spends a `CreateStakePool` utxo, this means the user is
+                            // is decommissioning the pool.
+                            let res = self
+                                .accounting_delta_adapter
+                                .operations(tx_source)
+                                .decommission_pool(*pool_id)
+                                .map_err(ConnectTransactionError::PoSAccountingError);
+                            Some(res)
+                        }
+                        TxOutput::ProduceBlockFromStake(_, pool_id) => {
+                            // If the input spends a `ProduceBlockFromStake` utxo, this means the user is
+                            // is decommissioning the pool.
+                            let res = self
+                                .accounting_delta_adapter
+                                .operations(tx_source)
+                                .decommission_pool(*pool_id)
+                                .map_err(ConnectTransactionError::PoSAccountingError);
+                            Some(res)
+                        }
+                        TxOutput::CreateDelegationId(_, _)
+                        | TxOutput::Transfer(_, _)
+                        | TxOutput::LockThenTransfer(_, _, _)
+                        | TxOutput::Burn(_) => None,
+                    },
+                    None => Some(Err(ConnectTransactionError::MissingOutputOrSpent)),
+                },
+                Err(_) => Some(Err(ConnectTransactionError::UtxoError(
+                    utxo::Error::ViewRead,
+                ))),
+            })
+            .collect::<Result<Vec<_>, _>>()?;
 
-        let tx_undo = tx
+        // Process tx outputs in terms of pos accounting.
+        let outputs_undos = tx
             .outputs()
             .iter()
             .filter_map(|output| match output {
-                TxOutput::CreateStakePool(data) => {
+                TxOutput::CreateStakePool(pool_id, data) => {
+                    let expected_pool_id = pos_accounting::make_pool_id(input0.outpoint());
+                    if expected_pool_id != *pool_id {
+                        Some(Err(ConnectTransactionError::SpendStakeError(
+                            SpendStakeError::StakePoolIdMismatch(expected_pool_id, *pool_id),
+                        )))
+                    } else {
+                        let res = self
+                            .accounting_delta_adapter
+                            .operations(tx_source)
+                            .create_pool(input0.outpoint(), data.as_ref().clone().into())
+                            .map(|(_, undo)| undo)
+                            .map_err(ConnectTransactionError::PoSAccountingError);
+                        Some(res)
+                    }
```

Comment:
Tiny style nitpick: maybe factor out the `Some` out of the `if` expression:

```rust
if expected != pool_id {
    Some(Err(err_result))
} else {
    let res = result;
    Some(res)
}
```

change to:

```rust
let res = if expected != pool_id {
    Err(err_result)
} else {
    result
};
Some(res)
```

Also the double negation incurs a slight additional cognitive load, I'd go just for an equality condition and switch the branches around:

```rust
let res = if expected == pool_id {
    result
} else {
    Err(err_result)
};
Some(res)
```

Hope I did not make a mistake somewhere